### PR TITLE
Shopping list displaying properly

### PR DESCRIPTION
**Pull request by krogowsk531**
# Matthew, KJ, Katy

- [x] checked functionality of the webpage
- [x] received help from a mentor/collaborator?
- [x] added teammates as reviewers

## Notes
* Wrote implementation code to make a shopping list of needed items now appear.

## Issues Addressed
* Addresses #49

## Questions
* n/a

## Problems or Challenges
* n/a

## Screenshots (if applicable)
* n/a

### CODE REVIEW

File: src/scripts.js [hunk 1]
```diff
@@ -221,11 +216,30 @@ function openRecipeInfo(event) {
...
-  domUpdates.generateRecipeTitle(recipe, generateIngredients(recipe, ingredientsData), fullRecipeInfo);
+  domUpdates.generateRecipeTitle(recipe, generateIngredients(recipe, ingredientsData), fullRecipeInfo, ingredientsData);
   domUpdates.addRecipeImage(recipe);
   domUpdates.generateInstructions(recipe, fullRecipeInfo);
   domUpdates.addOverlay(fullRecipeInfo);
+  let userShoppingList = document.querySelector('.recipe-instructions')
+  let htmlString = "<button class='cook-me'>Cook Me</button>"
+  userShoppingList.innerHTML += htmlString
+  let cookMeBtn = document.querySelector('.cook-me')
+  cookMeBtn.addEventListener('click', function() {
```
Comment (GreyMatteOr) [inline src/scripts.js:229]: Let's make sure to add any new Dom functionality into the Dom Updates file

File: src/scripts.js [hunk 2]
```diff
@@ -221,11 +216,30 @@ function openRecipeInfo(event) {
...
+    checkPantryIngredients(recipe, user.pantry)
+  })
+}
+
+function checkPantryIngredients(recipe, pantry) {
+  let missingIngredients = pantry.hasEnoughIngredients(recipe)
+  let userShoppingList = document.querySelector('.recipe-instructions')
+  missingIngredients.forEach(ingredient => {
+    console.log("KJ", ingredient)
+    let ingredientName = ingredientsData.find(i => i.id == Object.keys(ingredient)[0]).name
```
Comment (GreyMatteOr) [inline src/scripts.js:240]: UGH This is hideous. Let me rework the Pantry.js class so that the return data is easier to use. Sorry everyone!

**Review by kathrynljackson — APPROVED** [review]
So many changes! Looks great @krogowsk531. I'll start reformatting the button.

**Review by GreyMatteOr — APPROVED** [review]
So glad we're finalyl adding stuff instead of just changing BOB'S website

**Pull request merged**